tests/test_interpolation.py

```diff
@@ -211,71 +211,203 @@ def test_type_inherit_type(cfg: Any) -> None:
     assert type(cfg.s) == str  # check that string interpolations are always strings
 
 
+@pytest.mark.parametrize("env_func", ["env", "oc.env"])
+class TestEnvInterpolation:
+    @pytest.mark.parametrize(
+        "cfg,env_name,env_val,key,expected",
+        [
+            pytest.param(
+                {"path": "/test/${${env_func}:foo}"},
+                "foo",
+                "1234",
+                "path",
+                "/test/1234",
+                id="simple",
+            ),
+            pytest.param(
+                {"path": "/test/${${env_func}:not_found,ZZZ}"},
+                None,
+                None,
+                "path",
+                "/test/ZZZ",
+                id="not_found_with_default",
+            ),
+            pytest.param(
+                {"path": "/test/${${env_func}:not_found,a/b}"},
+                None,
+                None,
+                "path",
+                "/test/a/b",
+                id="not_found_with_default",
+            ),
+        ],
+    )
+    def test_env_interpolation(
+        self,
+        monkeypatch: Any,
+        env_func: str,
+        cfg: Any,
+        env_name: Optional[str],
+        env_val: str,
+        key: str,
+        expected: Any,
+    ) -> None:
+        if env_name is not None:
+            monkeypatch.setenv(env_name, env_val)
+
+        cfg["env_func"] = env_func  # allows choosing which env resolver to use
+        cfg = OmegaConf.create(cfg)
+
+        # The legacy env resolver triggers a deprecation warning.
+        if env_func == "env":
+            with pytest.warns(UserWarning):
+                assert OmegaConf.select(cfg, key) == expected
+        else:
+            assert OmegaConf.select(cfg, key) == expected
```

Comment:
You can add recwarn to the function signature to ignore all deprecation warnings from this function.

Reply:
Oh, nice! Done in a973d21